aqandu/api_routes.py

```diff
@@ -161,6 +163,193 @@ def liveSensors():
     return jsonify(sensor_list)
 
 
+@app.route("/api/getEstimateMap", methods=["GET"])
+def getEstimateMap():
+
+    # this species grid positions should be interpolated in UTM coordinates
+    # right now (Nov 2020) this is not supported.
+    # might be used later in order to build grids of data in UTM coordinates -- this would depend on what the display/visualization code needs
+    # after investigation, most vis toolkits support lat-lon grids of data. 
+    if "UTM" in request.args:
+        UTM = True
+    else:
+        UTM = False
+
+    # Get the arguments from the query string
+    if not UTM:
+        try:
+            lat_hi = float(request.args.get('lat_hi'))
+            lat_lo = float(request.args.get('lat_lo'))
+            lon_hi = float(request.args.get('lon_hi'))
+            lon_lo = float(request.args.get('lon_lo'))
+        except ValueError:
+            return 'lat, lon, lat_res, be floats in the lat-lon (not UTM) case', 400
+        try:
+            lat_size = int(request.args.get('lat_size'))
+            lon_size = int(request.args.get('lon_size'))
+        except ValueError:
+            return 'lat, lon, sizes must be ints (not UTM) case', 400
+
+        lat_res = (lat_hi-lat_lo)/float(lat_size)
+        lon_res = (lon_hi-lon_lo)/float(lon_size)
+
+    query_date = request.args.get('date')
+    if not utils.validateDate(query_date):
+        msg = f"Incorrect date format, should be {utils.DATETIME_FORMAT}, e.g.: 2018-01-03T20:00:00Z"
+        return msg, 400
+
+    query_datetime = utils.parseDateString(query_date)
+
+    print((
+        f"Query parameters: lat_lo={lat_lo} lat_hi={lat_hi}  lon_lo={lon_lo} lon_hi={lon_hi} lat_res={lat_res} lon_res={lon_res} date={query_datetime}"
+    ))
+
```

Comment:
Mostly removed in the new version.  I have left in most of the ones that were in place already.  These kinds of "reporting" print statements should be converted, eventually, into logs so that we can keep track of the queries that people are making.   If you want to comment them all out, then feel free to do so.  However, I would suggest leaving the code there because it offers good ideas on how to log this stuff.